app/lib/features/events/pages/event_details_page.dart

```diff
@@ -0,0 +1,412 @@
+          child: Center(child: CircularProgressIndicator()),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildEventAppBar(CalendarEvent calendarEvent) {
+    return Consumer(
+      builder: (context, ref, child) {
+        return SliverAppBar(
+          expandedHeight: 200.0,
+          pinned: true,
+          actions: [_buildActionMenu(calendarEvent)],
+          flexibleSpace: Container(
+            padding: const EdgeInsets.only(top: 20),
+            decoration: const BoxDecoration(gradient: primaryGradient),
+            child: const FlexibleSpaceBar(
+              background: Icon(Atlas.calendar_dots, size: 80),
+            ),
+          ),
+        );
+      },
+    );
+  }
+
+  Widget _buildActionMenu(CalendarEvent event) {
+    //Get membership details
+    final spaceId = event.roomIdStr();
+    final membership = ref.watch(roomMembershipProvider(spaceId));
+
+    //Create event actions
+    List<PopupMenuEntry> actions = [];
+
+    if (membership.valueOrNull != null) {
+      final member = membership.requireValue!;
+
+      //Edit Event Action
+      if (member.canString('CanPostEvent')) {
+        actions.add(
+          PopupMenuItem(
+            onTap: () => context.pushNamed(
+              Routes.editCalendarEvent.name,
+              pathParameters: {'calendarId': widget.calendarId},
+            ),
+            child: const Row(
+              children: <Widget>[
+                Icon(Atlas.pencil_edit_thin),
+                SizedBox(width: 10),
+                Text('Edit Event'),
+              ],
+            ),
+          ),
+        );
+      }
+
+      //Delete Event Action
+      if (member.canString('CanRedactOwn') &&
+          member.userId().toString() == event.sender().toString()) {
+        final roomId = event.roomIdStr();
+        actions.addAll([
+          PopupMenuItem(
+            onTap: () => showAdaptiveDialog(
+              context: context,
+              builder: (context) => RedactContentWidget(
+                title: 'Remove this post',
+                eventId: event.eventId().toString(),
+                onSuccess: () {
+                  ref.invalidate(calendarEventProvider);
+                  if (context.mounted) {
+                    context.goNamed(
+                      Routes.spaceEvents.name,
+                      pathParameters: {'spaceId': roomId},
+                    );
+                  }
+                },
+                senderId: event.sender().toString(),
+                roomId: roomId,
+                isSpace: true,
+              ),
+            ),
+            child: Row(
+              children: <Widget>[
+                Icon(
+                  Atlas.trash_can_thin,
+                  color: Theme.of(context).colorScheme.error,
+                ),
+                const SizedBox(width: 10),
+                const Text('Remove Event'),
+              ],
+            ),
+          ),
+        ]);
+      }
+    }
+
+    //Report Event Action
+    else {
```

Comment:
please, for clarity.
```suggestion
    } else {
    //Report Event Action
```